backend/src/routes/tag.ts

```diff
@@ -39,6 +38,45 @@ tagRouter.get("/", async (c) => {
 	}
 });
 
+tagRouter.get("/recommended", async (c) => {
+	try {
+		const prisma = new PrismaClient({
+			datasourceUrl: c.env.DATABASE_URL,
+		}).$extends(withAccelerate());
+
+		const topTags = await prisma.tagsOnPost.groupBy({
+			by: ["tagId"],
+			_count: {
+				postId: true
+			}, 
+			orderBy: {
+				_count: {
+					postId: 'desc',
+				}
+			},
+			take: 5
+		});
+
+		const topFiveTagsIds = topTags.map((t) => t.tagId)
+		const topfiveTags = await prisma.tag.findMany({
+			where: {
+				id: {
+					in: topFiveTagsIds
+				}
+			}
+		});
+		return c.json({
+			topfiveTags
+		});
+	} catch (error) {
+		console.log(error)
+		c.status(411);
+		return c.json({
+			message: "Something went wrong while fetching tags."
+		});
+	}
+});
```

Comment:
Instead of running 2 queries, can we do it in 1 ?

Something like this: (haven't tested it, consider it as a pseudo)
```javascript
		const topTags = await prisma.tag.findMany({
			orderBy: {
				tagsOnPost: {
					_count: 'desc',
				}
			},
			take: 5,
			include: {
				tagsOnPost: true,
			}
		});
```

Reply:
I think, this won't work.

Reply:
We can do something like:
```
const topTagsWithTagsOnPostCount = await prisma.tag.findMany({
			include: { 
				_count:{
					select: {
						tagsOnPost: true
					}
				}
			},
		});
```
to get:
![image](https://github.com/aadeshkulkarni/figuringout/assets/121121079/caf7a5f4-65e8-42f7-8962-b79c72a82156)

then sort list as per _count.tagOnPost.
But as the findMany does not support orderby '_count.tagOnPost' we can't fetch only the top 5 row. So, we will have to fetch all rows then sort them